scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerBackupRunner.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.core;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.apache.log4j.Logger;
+import org.joda.time.DateTime;
+import org.ow2.proactive.core.properties.PASharedProperties;
+import org.ow2.proactive.scheduler.common.util.ZipUtils;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+
+
+public class SchedulerBackupRunner implements Runnable {
+
+    private static final Logger LOGGER = Logger.getLogger(SchedulerBackupRunner.class);
+
+    private static final String PREFIX = "ProActiveSchedulerBackup_";
+
+    private final List<String> targets;
+
+    private final String destination;
+
+    private final int windowSize;
+
+    private final SchedulingService scheduler;
+
+    public SchedulerBackupRunner(SchedulingService scheduler) {
+        this.scheduler = scheduler;
+        targets = Arrays.asList(PASharedProperties.SERVER_BACKUP_TARGETS.getValueAsString().split(","));
+        destination = PASharedProperties.SERVER_BACKUP_DESTINATION.getValueAsString();
+        windowSize = PASharedProperties.SERVER_BACKUP_WINDOWS.getValueAsInt();
+    }
+
+    @Override
+    public void run() {
+        try {
+            scheduler.freeze();
+
+            while (!noOtherTaskIsRunning()) {
+                Thread.sleep(10000);
+            }
+            removeOldBackups();
+            performBackup();
+        } catch (IOException | InterruptedException e) {
+            LOGGER.error(e);
```

Comment:
LOGGER.error("", e) otherwise only the exception message is printed which is often empty